### PR TITLE
sql: add version field to PrivilegeDescriptor

### LINKED ISSUE #51930: sql: add FormatVersion field to PrivilegeDescriptor
After adding ownership to PrivilegeDescriptors, to be able to validate ownership, we have to know the version of the PrivilegedDescriptor. 

Versions before the addition of the ownership field are allowed to have an empty owner field whereas versions after should have owners.

### PR DESCRIPTION
**Pull request by RichardJCai**
Add version field to PrivilegeDescriptor to allow for
validating the owner field.

PrivilegeDescriptors pre 20.1 will have InitialVersion by default
and validation will not be performed for the owner field.

PrivilegeDescriptors 20.2 and onward will have OwnerVersion
and validation will be performed on the owner field.

Release note: None

Fixes https://github.com/cockroachdb/cockroach/issues/51930

### CODE REVIEW

File: pkg/sql/catalog/descpb/privilege.proto [hunk 1]
```diff
@@ -28,4 +28,6 @@ message PrivilegeDescriptor {
   option (gogoproto.equal) = true;
   repeated UserPrivileges users = 1 [(gogoproto.nullable) = false];
   optional string owner = 2 [(gogoproto.nullable) = false];
+  optional uint32 version = 3 [(gogoproto.nullable) = false,
+                                     (gogoproto.casttype) = "PrivilegeDescVersion"];
```
Comment (rohany) [inline pkg/sql/catalog/descpb/privilege.proto:32]: wierd spacing here

**Review by rohany — APPROVED** [review]
Make sure to audit all places that the privilege descriptor is created

**Comment by RichardJCai** [comment]
> Make sure to audit all places that the privilege descriptor is created

Yep, almost missed one in `newCommentPrivilegeDescriptor`

**Comment by RichardJCai** [comment]
bors r=rohany

**Pull request merged**